plugins/workspace-minimap/src/minimap.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A minimap is a miniature version of your blocks that
+ * appears on top of your main workspace. This gives you an overview
+ * of what your code looks like, and how it is organized.
+ * @author cesarades@google.com (Cesar Ades)
+ */
+
+import * as Blockly from 'blockly/core';
+
+// Events that should be send over to the minimap from the primary workspace
+const BLOCK_EVENTS_ = [
+    Blockly.Events.BLOCK_CHANGE,
+    Blockly.Events.BLOCK_CREATE,
+    Blockly.Events.BLOCK_DELETE,
+    Blockly.Events.BLOCK_DRAG,
+    Blockly.Events.BLOCK_MOVE];
+
+/**
+ * A minimap is a miniature version of your blocks that appears on
+ * top of your main workspace. This gives you an overview of what
+ * your code looks like, and how it is organized.
+ * @param {!Blockly.WorkspaceSvg} workspace The workspace to sit in.
+ */
+export class Minimap {
+    protected workspace_: Blockly.WorkspaceSvg;
+    protected minimap_: Blockly.WorkspaceSvg;
+    /**
+     * Constructor for a minimap
+     * @param {!Blockly.WorkspaceSvg} workspace The workspace to sit in.
+     */
+    constructor(workspace: Blockly.WorkspaceSvg) {
+        this.workspace_ = workspace;
+        const options = {
+            readOnly: true,
+        }
+        this.minimap_ = Blockly.inject('minimapDiv', options);
+        this.mirror = this.mirror.bind(this);
+    }
+  
+    /**
+     * Initialize.
+     */
+    init(): void {
+        this.workspace_.addChangeListener(this.mirror);
+    }
+
+    /**
+     * Creates the mirroring between workspaces. Passes on all desired events
+     * to the minimap from the primary worskpace.
+     * @param {Blockly.Events.Abstract} event The primary workspace event.
+     * @protected
+     */
+    mirror(event: Blockly.Events.Abstract): void {
+        if (BLOCK_EVENTS_.indexOf(event.type) > -1) {
```

Comment:
If you just want to see if the event type is one of the "approved" ones, you can also use a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set).

Reply:
The set was not working since it checks if the contents reference the same object

Reply:
Hmm it should work if you use the type strings (like you're currently doing). E.g: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/has#using_the_has_method

Reply:
Yea I tried that idk